resdk/resolwe.py

```diff
@@ -293,16 +293,23 @@ def run(self, slug=None, input={}, descriptor=None,  # pylint: disable=redefined
         else:
             raise ValueError("Unexpected behaviour at get process with slug {}".format(slug))
 
-        # Upload files in basic:file fields
+        # Pre-process inputs
         try:
             for schema, fields in iterate_fields(input, process['input_schema']):
                 field_name = schema['name']
                 field_type = schema['type']
                 field_value = fields[field_name]
 
+                # Wrapp `list:` fields into list if they are not already
```

Comment:
I had the same thoughts, but came to the conclusion that user probably doesn't care what is happening in the background as long as things work and are nice to use.
There is really ugly exception if you try to run `res.run('upload-fastq-single', input={'src': 'reads.fastq.gz'})` only because you forgot to add the brackets.